homeassistant/components/hive/switch.py

```diff
@@ -88,4 +64,7 @@ async def async_update(self):
         """Update all Node data from Hive."""
         await self.hive.session.updateData(self.device)
         self.device = await self.hive.switch.getSwitch(self.device)
-        self.attributes.update(self.device.get("attributes", {}))
+        self._attr_available = self.device["deviceData"].get("online")
+        if self._attr_available:
```

Comment:
I don't think we need this check.

Reply:
I have tested this through and the issue is around the "status" key doesn't exist in the dictionary when the device is offline. I'm not sure if when home assistant migrated to using _attr_ declarations they changed some logic around this as you have mentioned it did work previously without the check but the check is required now.

Reply:
Ok.